### PR TITLE
Introduce scrollIntoView for WebDriver & Appium

### PR DESCRIPTION
**Pull request by pablopaul**
See https://webdriver.io/docs/api/element/scrollIntoView.html

### CODE REVIEW

File: lib/helper/Appium.js [hunk 1]
```diff
@@ -1350,7 +1350,16 @@ class Appium extends Webdriver {
     if (this.isWeb) return super.grabValueFrom(locator);
     return super.grabValueFrom(parseLocator.call(this, locator));
   }
-
+  
+  /**
+   * {{> scrollIntoView }}
```
Comment (PeterNgTr) [inline lib/helper/Appium.js]: may you add the documentation for this?

File: lib/helper/Appium.js [hunk 2]
```diff
@@ -1350,7 +1350,16 @@ class Appium extends Webdriver {
...
+   *
+   * * Supported only for web testing
+   */
+  async scrollIntoView(locator) {
+    if (this.isWeb) return super.scrollIntoView(locator);
+  }
+
```
[the remark this answers is not in the document]
Comment (pablopaul) [reply lib/helper/Appium.js]: @PeterNgTr thanks

File: lib/helper/Appium.js (hunk 1 above)
Comment (pablopaul) [reply lib/helper/Appium.js]: @PeterNgTr added docu here: docs/webapi/scrollIntoView.mustache

File: docs/webapi/scrollIntoView.mustache [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Scroll element into viewport.
+
+```js
+I.scrollIntoView('#submit');
```
Comment (PeterNgTr) [inline docs/webapi/scrollIntoView.mustache]: it's better to add the examples with using params

Comment (pablopaul) [reply docs/webapi/scrollIntoView.mustache]: thanks @PeterNgTr - added

File: lib/helper/WebDriver.js [hunk 4]
```diff
@@ -1312,6 +1312,17 @@ class WebDriver extends Helper {
     return this.browser.executeAsync.apply(this.browser, arguments);
   }
 
+  /**
+   * {{> scrollIntoView }}
+   *
+   */
+  async scrollIntoView(locator, scrollIntoViewOptions) {
```
Comment (PeterNgTr) [inline lib/helper/WebDriver.js]: can we add tests @pablopaul?

Comment (pablopaul) [reply lib/helper/WebDriver.js]: @PeterNgTr what would be your approach? Could you outline your steps for me? I try to follow them :)

Comment (PeterNgTr) [reply lib/helper/WebDriver.js]: https://github.com/Codeception/CodeceptJS/blob/master/test/helper/WebDriver_test.js

so, I think you can add tests in here 👍

File: lib/helper/Appium.js [hunk 5]
```diff
@@ -1351,6 +1351,15 @@ class Appium extends Webdriver {
     return super.grabValueFrom(parseLocator.call(this, locator));
   }
 
+  /**
+   * {{> scrollIntoView }}
+   *
+   * Supported only for web testing
+   */
+  async scrollIntoView(locator, scrollIntoViewOptions) {
```
Comment (Vorobeyko) [inline lib/helper/Appium.js]: If `scrollIntoView` may be used only in web, why add this method for Appium? 
Maybe delete this method from Appium?

Comment (pablopaul) [reply lib/helper/Appium.js]: "web testing" means Appium web testing here @voro

**Comment by Vorobeyko** [comment]
Can you generate docs and typing please? 
Run:
```sh
npm run docs
npm run def
```

**Comment by pablopaul** [comment]
@Vorobeyko docs are added, for `npm run def` I get the following error:

```
npm run def

> codeceptjs@2.3.5 def /Users/paul/projects/CodeceptJS
> jsdoc -c typings/jsdoc.conf.js

[TSD-JSDoc] Failed to find parent of doclet 'module.exports' using memberof 'module', this is likely due to invalid JSDoc.
[TSD-JSDoc] Failed to find parent of doclet 'container.mocha' using memberof 'container', this is likely due to invalid JSDoc.
[TSD-JSDoc] Failed to find parent of doclet 'module.exports' using memberof 'module', this is likely due to invalid JSDoc.
[TSD-JSDoc] Failed to find parent of doclet 'context.Scenario' using memberof 'context', this is likely due to invalid JSDoc.
[TSD-JSDoc] Unable to resolve type for an unnamed item, this is likely due to invalid JSDoc. Often this is caused by invalid JSDoc on a parameter. Defaulting to `any`.
[TSD-JSDoc] Unable to resolve type for an unnamed item, this is likely due to invalid JSDoc. Often this is caused by invalid JSDoc on a parameter. Defaulting to `any`.
[TSD-JSDoc] Unable to resolve type for an unnamed item, this is likely due to invalid JSDoc. Often this is caused by invalid JSDoc on a parameter. Defaulting to `any`.
```

File: lib/helper/WebDriver.js (hunk 4 above)
Comment (pablopaul) [reply lib/helper/WebDriver.js]: @PeterNgTr I added a test

**Comment by DavertMik** [comment]
I approved. The next person who approves this should merge it

**Comment by pablopaul** [comment]
@PeterNgTr @DavertMik I rebased on latest master to resolve typings merge conflict, therefor I need your approval again.

**Pull request merged**